doc/stdenv/platform-notes.chapter.md

```diff
@@ -59,79 +57,59 @@ Some common issues when packaging software for Darwin:
 
   ```nix
   stdenv.mkDerivation {
     name = "libfoo-1.2.3";
-    prePatch = ''
-      substituteInPlace Makefile \
-        --replace-fail /usr/bin/xcrun xcrun
-        # or: --replace-fail /usr/bin/xcrun '${lib.getExe' buildPackages.xcbuild "xcrun"}'
-    '';
+    xcbuildFlags = [
+      "-configuration"
+      "Release"
+      "-project"
+      "libfoo-project.xcodeproj"
+      "-scheme"
+      "libfoo Package (macOS only)"
+    ];
+    __structuredAttrs = true;
   }
   ```
 
-- Multiple SDKs are available for use in nixpkgs. Each platform has a default SDK (10.12.2 for x86_64-darwin and 11.3 for aarch64-darwin), which is available as the `apple-sdk` package.
+#### Package requires a non-default SDK or fails to build due to missing frameworks or symbols
 
-  The SDK provides the necessary headers and text-based stubs to link common frameworks and libraries (such as libSystem, which is effectively Darwin’s libc). Projects will sometimes indicate which SDK to use by the Xcode version. As a rule of thumb, subtract one from the Xcode version to get the available SDK in nixpkgs.
+In some cases, you may have to use a non-default SDK. This can happen when a package requires APIs that are not present in the default SDK. For example, Metal Performance Shaders were added in macOS 12. If the default SDK is 11.3, then a package that requires Metal Performance Shaders will fail to build due to missing frameworks and symbols.
 
-  The `DEVELOPER_DIR` variable in the build environment has the path to the SDK in the build environment. The `SDKROOT` variable there contains a sysroot with the framework, header, and library paths. You can reference an SDK’s sysroot from Nix using the `sdkroot` attribute on the SDK package. Note that it is preferable to use `SDKROOT` because the latter will be resolved to the highest SDK version of any available to your derivation.
+To use a non-default SDK, add it to your derivation’s `buildInputs`. If your derivation needs a non-default SDK at build time (e.g., for a `depsBuildBuild` compiler), see the cross-compilation documentation for which input you should use.
 
-  ```nix
-  stdenv.mkDerivation {
-    name = "libfoo-1.2.3";
-    # ...
-    env.PACKAGE_SPECIFIC_SDK_VAR = apple-sdk_10_15.sdkroot;
-    # or
-    buildInputs = [ apple-sdk_10_15 ];
-    postPatch = ''
-      export PACKAGE_SPECIFIC_SDK_VAR=$SDKROOT
-    '';
-  }
-  ```
+When determining whether to use a non-default SDK, consider the following:
 
-  The following is a list of Xcode versions, the SDK version in nixpkgs, and the attribute to use to add it. Generally, only the last SDK release for a major version is packaged (each _x_ in 10._x_ until 10.15 is considered a major version).
+- Try building your derivation with the default SDK. If it works, you’re done.
+- If the package specifies a specific version, use that. See below for how to map Xcode version to SDK version.
```

Comment:
Maybe some hints for the most common ways/places a project defines the SDK version?

Reply:
I added some suggestions about looking in the supported platforms or installation documentation for the package.

Reply:
I’m a bit confused as to the overlap between this section and the later “Picking an SDK version”, now that this gives advice about… well… picking an SDK version – perhaps they could be merged?

Reply:
I kept the suggestion for where to find it but removed the wider commentary. I think it’s important to have the table be its own section, so I don’t want to combine it with the other one.